src/main/java/com/nirmata/workflow/details/internalmodels/StartedTask.java

```diff
@@ -39,6 +41,16 @@ public LocalDateTime getStartDateUtc()
     {
         return startDateUtc;
     }
+    
+    public int getProgress()
+    {
+        return progress;
+    }
+    
+    public void setProgress(int progress)
```

Comment:
The reason of this patch is to add progress field into StartedTask and allow TaskExecutor to update this field during a task's execution. That's why a setter is added. Whenever a task wants to update its progress, we deserialize the task from ZK, update the progress field then serialize it back.